components/global/LanguageSelector.vue

```diff
@@ -1,17 +1,22 @@
 <script setup>
-const { locale } = useI18n(); // eslint-disable-line
+const { toggleLocale } = useAppLocale();
```

Comment:
1. The line is there because else it would fail in production build
2. We don't need a composables for this. 

I suggest this instead:

```js
// if we remove this line even if it's not used directly, it won't build in production properly.
const { locale } = useI18n(); // eslint-disable-line
// update local storage upon language change
const toggleLocale = () => {
  localStorage.setItem("nuxt-lang", locale.value);
};

// check for stored language on initial load.
onMounted(() => {
  const storedLanguage = localStorage.getItem("nuxt-lang");
  if (storedLanguage) {
    locale.value = storedLanguage;
  }
});
```